src/LoginApprover.ts

```diff
@@ -20,7 +20,7 @@ export default class LoginApprover {
 	private _handler: AuthenticationClient;
 
 	constructor(accessToken: string, sharedSecret: string|Buffer, options?: ConstructorOptions) {
-		let agent:HTTPS.Agent = new HTTPS.Agent({keepAlive: true});
+		let agent:HTTPS.Agent = options.agent || new HTTPS.Agent({keepAlive: true});
 		if (options.httpProxy && options.socksProxy) {
 			throw new Error('Cannot specify both httpProxy and socksProxy at the same time');
```

Comment:
I'd like to see this check updated to enforce that only one of `httpProxy`, `socksProxy`, or `agent` is used.

Reply:
> I'd like to see this check updated to enforce that only one of `httpProxy`, `socksProxy`, or `agent` is used.

Done